### PR TITLE
Move all E2E test templates to use external CCM and CSI

### LINKED ISSUE #2941: [e2e] Enable out-of-tree cloud-provider and CSI in e2e tests
Since we are getting close to the release where in-tree cloud providers are removed from Kubernetes repo, it would be good to enable out-of-tree cloud providers by default in all of our e2e tests.

This gets us closer to defaulting to out-of-tree cloud provider issue is: https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/964

/area testing
/priority important-soon
/milestone v1.x
/triage accepted

### PR DESCRIPTION
**Pull request by wyike**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #2941 

**Special notes for your reviewer**:

Remaining work:
1. bump CCM and CSI version
2. Update templates under test/e2e/data/infrastructure-aws/withoutclusterclass/e2e_test_templates and test/e2e/data/infrastructure-aws/withclusterclass/e2e_test_templates, after this PR merged
3. Note: one testcase using capi-upgrades https://github.com/kubernetes-sigs/cluster-api-provider-aws/tree/main/test/e2e/data/infrastructure-aws/capi-upgrades/v1beta1 is still using in tree ccm as the upgrade-from cluster template. (I cannot recall the exact test case name, but I remember there is such one)
4. Didn't take care `PDescribe` testcases

**Checklist**:

- [x] squashed commits
- [ ] includes documentation
- [ ] adds unit tests
- [x] adds or updates e2e tests

### CODE REVIEW

File: test/e2e/data/e2e_conf.yaml [hunk 1]
```diff
@@ -98,7 +98,7 @@ providers:
           - sourcePath: "./infrastructure-aws/withoutclusterclass/generated/cluster-template-efs-support.yaml"
           - sourcePath: "./infrastructure-aws/withoutclusterclass/generated/cluster-template-external-csi.yaml"
           - sourcePath: "./infrastructure-aws/withoutclusterclass/generated/cluster-template-csimigration-off.yaml"
-          - sourcePath: "./infrastructure-aws/withoutclusterclass/generated/cluster-template-external-cloud-provider.yaml"
+          - sourcePath: "./infrastructure-aws/withoutclusterclass/generated/cluster-template-intree-cloud-provider.yaml"
```
Comment (Ankitasw) [inline test/e2e/data/e2e_conf.yaml:101]: Any reason to change it to intree. This test uses external CSI and external CCM.

File: test/e2e/data/infrastructure-aws/withoutclusterclass/kustomize_sources/default/kustomization.yaml [hunk 2]
```diff
@@ -1,6 +1,25 @@
+# default template enables external ccm and csi
```
Comment (wyike) [inline test/e2e/data/infrastructure-aws/withoutclusterclass/kustomize_sources/default/kustomization.yaml:1]: Hi @Ankitasw, 
Default template enables external ccm and csi. Templates for other flavors are based on this default template and then patch some more.
So we don't need the external-cloud-provider template but we need an intree-cloud-provider template for the CCM/CSI upgrade tests.

Comment (Ankitasw) [reply test/e2e/data/infrastructure-aws/withoutclusterclass/kustomize_sources/default/kustomization.yaml:1]: Got it 👍

**Comment by Ankitasw** [comment]
cc @Skarlso

**Pull request merged**